### ISSUE TITLE
Bug: TypeScript version warning shows with TS 5.5

### ISSUE THREAD

**Issue by cm-ayf** [typescript-eslint/typescript-eslint#9399]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Repro Code

```TypeScript
{
  "name": "typescript-eslint-bug-repro",
  "version": "0.0.0",
  "devDependencies": {
    "eslint": "8.57.0",
    "typescript": "5.5.2",
    "typescript-eslint": "7.13.1"
  }
}
```

### ESLint Config

```javascript
import eslint from "@eslint/js";
import tseslint from "typescript-eslint";

export default tseslint.config(...tseslint.configs.recommendedTypeChecked);
```

### tsconfig

```jsonc
{}
```

### Expected Result

running `eslint .` does not show any warning on SUPPORTED TYPESCRIPT VERSIONS

### Actual Result

running `eslint .` does show warning `SUPPORTED TYPESCRIPT VERSIONS: >=4.7.4 <5.5.0`

### Additional Info

This issue occurs because #9397 did not update [`SUPPORTED_TYPESCRIPT_VERSIONS` in packages/typescript-estree/src/parseSettings/warnAboutTSVersion.ts](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-estree/src/parseSettings/warnAboutTSVersion.ts#L9). Updating this will resolve this issue. I will PR this.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `7.13.1` |
| `@typescript-eslint/parser`            | `7.13.1` |
| `@typescript-eslint/scope-manager`     | `7.13.1` |
| `@typescript-eslint/typescript-estree` | `7.13.1` |
| `@typescript-eslint/type-utils`        | `7.13.1` |
| `@typescript-eslint/utils`             | `7.13.1` |
| `TypeScript`                           | `7.13.1` |
| `ESLint`                               | `8.57.0` |
| `node`                                 | `20.14.0` |

**Comment by cm-ayf** [comment]
I have just noticed that #9397 is not released yet :D

**Comment by bradzacher** [comment]
It's not - but you caught something I'd forgotten anyways!

**Comment by voxpelli** [comment]
Notes for those who like me are on v8: Neither this or #9397 has been merged into the v8 branch, so the latest v8 nightly of the v8 branch does not support TS 5.5 yet

**Comment by hildjj** [comment]
Should this issue be re-opened for v8, or should there be a new issue to port the change?

**Comment by JoshuaKGoldberg** [comment]
Let's go with a new issue. 👍

**Comment by voxpelli** [comment]
I didn’t open a new issue as it looked like `main` has been merged into `v8` every now and then, and if anything that would be the issue I would create, to do a new sync
